### PR TITLE
Switch to go modules

### PR DESCRIPTION
**Pull request by spangenberg**
Switch to go modules, this is the second PR after https://github.com/openshift/cluster-api-actuator-pkg/pull/99 for switching to go modules within the cloud team.
CI PR will follow up in openshift/release, but obviously we first need to merge this one.

### CODE REVIEW

File: hack/verify-actuator-pkg.sh [hunk 1]
```diff
@@ -2,12 +2,5 @@
...
-if ! command -v dep 1>/dev/null 2>&1; then
-	curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-fi
-
-if dep check | grep -q cluster-api-actuator-pkg; then
-	exit 1
-fi
-dep ensure -update github.com/openshift/cluster-api-actuator-pkg
-git diff --exit-code
+go get -u github.com/openshift/cluster-api-actuator-pkg
```
Comment (michaelgugino) [inline hack/verify-actuator-pkg.sh]: Do we use this or still need this?  I think we're explicitly cloning this from master when we run tests?

File: tools.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// +build tools
```
Comment (michaelgugino) [inline tools.go]: Can we add a comment in this file as to why it's required and what we hope it achieves?  Pretty confusing stuff here.

File: hack/verify-actuator-pkg.sh (hunk 1 above)
Comment (spangenberg) [reply hack/verify-actuator-pkg.sh]: No, we're not. It's using the version which is marked in the go.mod/go.sum file.
We can discuss wether we always wanna use the latest version, but so far it's not.

Comment (bison) [reply hack/verify-actuator-pkg.sh]: This script no longer used. It's been removed from the CI config. We should delete it from the repos and remove the target in the Makefile. We can save that for another PR though.

File: tools.go (hunk 2 above)
Comment (spangenberg) [reply tools.go]: Done

File: hack/e2e.sh [hunk 3]
```diff
@@ -2,11 +2,16 @@
 
 set -euo pipefail
 
+GO111MODULE = off
```
Comment (bison) [inline hack/e2e.sh]: Can't have spaces here. These should read: `export GO111MODULE=off`

I'd recommend running `shellcheck` on the scripts here. Also, if we have to change this anyway, maybe we should switch to using `git clone` to grab the repo, and avoid having to explicitly override these here.

File: hack/e2e.sh [hunk 4]
```diff
@@ -2,11 +2,9 @@
 
 set -euo pipefail
 
-GOPATH="$(mktemp -d)"
-export GOPATH
+tmp="$(mktemp -d)"
+trap 'rm -rf "$tmp"' EXIT
```
Comment (bison) [inline hack/e2e.sh]: I'd personally prefer we drop this. This is almost exclusively used in CI where it doesn't matter if we clean this up, and we might want it to stick around so we can see what we built.

**Comment by ingvagabund** [comment]
```
$ make generate
go generate ./pkg/... ./cmd/...
W0819 15:53:59.471502   25296 parse.go:239] Ignoring child directory awsproviderconfig: unable to import "awsproviderconfig": go/build: importGo awsproviderconfig: exit status 1
can't load package: package awsproviderconfig: cannot find package "." in:
	/home/jchaloup/Projects/src/sigs.k8s.io/cluster-api-provider-aws/vendor/awsproviderconfig

W0819 15:53:59.497258   25296 parse.go:239] Ignoring child directory awsproviderconfig/v1beta1: unable to import "awsproviderconfig/v1beta1": go/build: importGo awsproviderconfig/v1beta1: exit status 1
can't load package: package awsproviderconfig/v1beta1: cannot find package "." in:
	/home/jchaloup/Projects/src/sigs.k8s.io/cluster-api-provider-aws/vendor/awsproviderconfig/v1beta1
```

**Comment by spangenberg** [comment]
@ingvagabund thanks, deepcopy-gen doesn't support go modules. Had to switch to controller-tools.

FYI https://github.com/kubernetes-sigs/controller-tools/issues/120

**Comment by enxebre** [comment]
hey @spangenberg this PR includes multiple commits undoing changes introduced by the PR itself, let's make sure we cleanup the git history once there's consensus to merge

**Pull request merged**